### PR TITLE
feat: add dropdown to form field options

### PR DESCRIPTION
**Pull request by realSpok**
## Definition of Done
#85yxm8nc0
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-toolkit/src/interfaces/action.ts [hunk 1]
```diff
@@ -10,35 +10,62 @@ export type File = {
...
+] as const;
+
+export type ActionFieldType = (typeof ActionFieldTypeList)[number];
+
+export type ActionFieldDropdown = ActionFieldBase & {
+  widget: 'Dropdown';
+  type: 'Date' | 'Dateonly' | 'Number' | 'String';
+  options?: DropdownOption[];
+  search?: 'static' | 'disabled';
+};
```
Comment (realSpok) [inline packages/datasource-toolkit/src/interfaces/action.ts:46]: This looks a lot better than previously

Comment (realSpok) [reply packages/datasource-toolkit/src/interfaces/action.ts:46]: It looks like you added types that will not be used in the first release, is that on purpose ?

Comment (ghusse) [reply packages/datasource-toolkit/src/interfaces/action.ts:46]: Which types?

File: packages/agent/src/utils/forest-schema/action-fields.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+} from '@forestadmin/datasource-toolkit';
+
+export default class ActionFields {
+  public static isCollectionField(
+    field: ActionField | null | undefined,
+  ): field is ActionFieldCollection {
+    return field?.type === 'Collection';
+  }
+
+  public static isEnumField(field: ActionField | null | undefined): field is ActionFieldEnum {
```
Comment (realSpok) [inline packages/agent/src/utils/forest-schema/action-fields.ts:16]: shouldn't we test for `EnumList` as well ?

File: packages/agent/src/utils/forest-schema/generator-action-field-widget.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    field: ActionField,
+  ): ForestServerActionFieldDropdownOptions | undefined {
+    if (!ActionFields.hasWidget(field) || ['Collection', 'Enum', 'EnumList'].includes(field.type))
+      return undefined;
+
+    switch (true) {
+      case ActionFields.isDropdownField(field):
+        return GeneratorActionFieldWidget.buildDropdownWidgetEdit(field);
+      default:
+        throw new Error(`Unsupported widget type: ${field.widget}`);
```
Comment (realSpok) [inline packages/agent/src/utils/forest-schema/generator-action-field-widget.ts:17]: I guess this is good enough for the moment

File: packages/datasource-toolkit/src/interfaces/action.ts (hunk 1 above)
Comment (realSpok) [reply packages/datasource-toolkit/src/interfaces/action.ts:47]: `search: 'static'` will not be supported at the moment for instance

File: packages/agent/src/utils/forest-schema/action-fields.ts (hunk 2 above)
Comment (realSpok) [reply packages/agent/src/utils/forest-schema/action-fields.ts:16]: ok, I saw that you have split it

**Comment by realSpok** [comment]
- ✅  Code review
  - ✅  Use pure functions when possible
  - ⤬  Performance concerns
  - ✅  Security concerns
  - ✅  The PR tackle one subject only
- ✅ Automatic tests
  - ✅  Unit tests
  - ✅  Integration tests
- ✅ Manual tests
  -  ✅  Test the functionality
  -  ✅  Test error cases
- ✅  PR title
- ✅  Destination branch
- ✅  PR linked to the clickup task

**Review by ghusse — APPROVED** [review]

**Pull request merged**